### DISCUSSION TITLE
GDS import to "3D modeler",  Not to EDB, Not to "3D-layout" but just to "3D Modeler"

### DISCUSSION THREAD

**Discussion by IkeChoice** [ansys/pyaedt#3697] [Q&A] [answered]
There are several reasons why our guys don't want to use "3D layout"
So, We used to import a gds file directly to the "3D modeler". 
I just wanted to change the code below using the pyaedt API

But I could not find out any similiar pyaedt API.
is anyone who can help me out how to import a GDS to "3D modeler" using PyAedt API?
Do I import to an EDB and convert to the "3D modeler"?

I would very appreciate it if anyone can give me any comments

 The ipython script to import GDS to HFSS 3D modeler - how do I convert to pyAedt world?
 ################################################################################
 ################################################################################
ScriptEnv.Initialize("Ansoft.ElectronicsDesktop")
oDesktop.RestoreWindow()
oProject = oDesktop.SetActiveProject("myDieProject")
oProject.InsertDesign("HFSS", "gds_imported", "HFSS Modal Network", "")
oDesign = oProject.SetActiveDesign("gds_imported")
**oEditor = oDesign.SetActiveEditor("3D Modeler") # please this see the oEditor is "3D modeler"**
oEditor.ImportGDSII(
[
		"NAME:options",
		"FileName:="	, "my_gds_file_path"
		"FlattenHierarchy:="	, True,
		[
			"NAME:LayerMap",
			[
                          ........................................
                          ......... Layer maps ...............
                         ................................................
                        ]
                       "OrderMap:="		, ["entry:="		, [	"..................................................................] ]
                ]
]

**Comment by IkeChoice** [comment]
It seems like I need to create a control file or xml file.

One thing that I'd like to comment is **the HFSS layer map** does not work with import_gds_file() 

According to the document HFSS help (2023R1),  the GDS layer maping file consists of **the import number, destinantion name, color, elevation, thickness** (please refer to 3-107 of  HFSS help, 2023R1)

if the layer map is used  with import_gds_file() then the list of  layer name, layer number and layer type is the expected input. 
(please look at ControlFile().parse_layer_map() 

I hope there is  the way to use **the HFSS layer map**

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @IkeChoice ,

AEDT Layout workflow is first to convert the ECAD format (gerber, odb++, dxf...) to Ansys layout format called EDB. Then once you have your EDB file, you can transform it to an HFSS, Q3D or Maxwell.

Using PyAEDT is just:

1) First, if you want to see in graphical mode the layout, you can import the gerber using this method:

https://aedt.docs.pyansys.com/version/0.6/API/_autosummary/pyaedt.hfss3dlayout.Hfss3dLayout.import_gerber.html

Then you can manipulate the Layout.

If you need intensive Layout manipulation, I encourage you to use PyAEDT EDB (much better):

https://aedt.docs.pyansys.com/version/0.6/EDBAPI/_autosummary/pyaedt.edb.Edb.html#pyaedt.edb.Edb

As you can see in the previous link, you can load the gerber with the technology file, and it will create an EDB file (Ansys native Layout format)

Once you have this object, you can perform some operations like cutouts, adding pins, removing components, etc... and finally, you can export the layout to Maxwell:

https://aedt.docs.pyansys.com/version/0.6/EDBAPI/_autosummary/pyaedt.edb.Edb.export_maxwell.html

Regarding the layer mapping, please check EDB examples, for example, the next one shows how to manipulate a GDS control file if needed:
https://aedt.docs.pyansys.com/version/0.6/examples/00-EDB/10_GDS_workflow.html#sphx-glr-examples-00-edb-10-gds-workflow-py

**Reply by IkeChoice** [reply]
Thank you much
I like it!
